allauth/socialaccount/providers/apple/provider.py

```diff
@@ -9,10 +9,23 @@
 from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
 
 
+class AppleAccount(ProviderAccount):
+    def to_str(self):
+        dflt = super().to_str()
+        email = self.account.extra_data.get("email")
+        name = self.account.extra_data.get("name", {})
+        if name and (name.get("firstName") or name.get("lastName")):
+            full_name = " ".join([name.get("firstName", ""), name.get("lastName", "")])
+            full_name = full_name.strip()
+            if full_name:
+                return f"{full_name} <{email or dflt}>"
```

Comment:
I am not sure the `<...>` approach is appropriate here. This is inconsistent compared to the other providers. Also, using `<dflt>` is perhaps unexpected.

Reply:
The email address provided by Apple may be a masked email address ("Hide My Email" feature), and so the user may not recognise the email address, or know which Apple account it belongs to. I'm not sure what the best identifier for an Apple account is.

Reply:
The email is masked if it ends with `@privaterelay.appleid.com`.  So we could check if the email is masked, and if not, return that as is, otherwise, return just the full name (as the email does not convey any information at all). Then, we wouldn't need the `<...>` construct.